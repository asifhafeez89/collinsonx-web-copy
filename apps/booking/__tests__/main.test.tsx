import * as React from 'react';
import renderer from 'react-test-renderer';
import Main from '../pages/index';

describe('<Home />', () => {
  it('renders outline variant', () => {
    const tree = renderer
      .create(
        <Main consumerNumber={''} tempBearerToken={''} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
