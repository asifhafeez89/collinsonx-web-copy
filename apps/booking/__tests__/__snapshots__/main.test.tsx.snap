// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders outline variant 1`] = `
[
  .emotion-2 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  color: inherit;
  font-size: inherit;
  line-height: 1.55;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-weight: 700;
  font-size: 2rem;
  line-height: 1.3;
  margin: 0;
  margin-bottom: 0.5rem;
}

.emotion-2:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

<h1
    className="emotion-0 emotion-1 emotion-2"
  >
    Welcome to Booking
  </h1>,
  <p>
    Consumer Number: 
  </p>,
  <p>
    Temporary Bearer Token: 
  </p>,
  .emotion-0 {
  margin-right: auto;
  margin-left: auto;
  max-width: 20rem;
}

.emotion-3 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
}

.emotion-6 {
  display: inline-block;
  font-size: 0.875rem;
  font-weight: 500;
  color: #212529;
  word-break: break-word;
  cursor: default;
  -webkit-tap-highlight-color: transparent;
}

.emotion-7 {
  color: #fa5252;
}

.emotion-12 {
  position: relative;
}

.emotion-15 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: 2.25rem;
  -webkit-tap-highlight-color: transparent;
  line-height: calc(2.25rem - 0.125rem);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 0.875rem;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  border: 0.0625rem solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  min-height: 2.25rem;
  padding-left: calc(2.25rem  / 3);
  padding-right: calc(2.25rem  / 3);
  border-radius: 0.25rem;
}

.emotion-15:focus,
.emotion-15:focus-within {
  outline: none;
  border-color: #228be6;
}

.emotion-15:disabled,
.emotion-15[data-disabled] {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
}

.emotion-15:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-15:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-15:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-15[data-disabled]::-webkit-input-placeholder {
  color: #909296;
}

.emotion-15[data-disabled]::-moz-placeholder {
  color: #909296;
}

.emotion-15[data-disabled]:-ms-input-placeholder {
  color: #909296;
}

.emotion-15:disabled::placeholder,
.emotion-15[data-disabled]::placeholder {
  color: #909296;
}

.emotion-15[data-invalid] {
  color: #fa5252;
  border-color: #fa5252;
}

.emotion-15[data-invalid]::-webkit-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-15[data-invalid]::-moz-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-15[data-invalid]:-ms-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-15[data-invalid]::placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-15[data-with-icon] {
  padding-left: 2.25rem;
}

.emotion-15::-webkit-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-15::-moz-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-15:-ms-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-15::placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-15::-webkit-inner-spin-button,
.emotion-15::-webkit-outer-spin-button,
.emotion-15::-webkit-search-decoration,
.emotion-15::-webkit-search-cancel-button,
.emotion-15::-webkit-search-results-button,
.emotion-15::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-15[type=number] {
  -moz-appearance: textfield;
}

.emotion-18 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  line-height: 1.55;
  margin-right: auto;
  margin-left: auto;
  max-width: 25rem;
}

.emotion-28 {
  pointer-events: none;
  position: absolute;
  z-index: 1;
  left: 0;
  top: 0;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 2.25rem;
  color: #adb5bd;
}

.emotion-34 {
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  height: auto;
  -webkit-tap-highlight-color: transparent;
  line-height: 1.55;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  resize: none;
  box-sizing: border-box;
  font-size: 0.875rem;
  width: 100%;
  color: #000;
  display: block;
  text-align: left;
  border: 0.0625rem solid #ced4da;
  background-color: #fff;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  min-height: 2.25rem;
  padding-left: calc(2.25rem  / 3);
  padding-right: calc(2.25rem  / 3);
  border-radius: 0.25rem;
  cursor: pointer;
  line-height: unset;
}

.emotion-34:focus,
.emotion-34:focus-within {
  outline: none;
  border-color: #228be6;
}

.emotion-34:disabled,
.emotion-34[data-disabled] {
  background-color: #f1f3f5;
  color: #909296;
  opacity: 0.6;
  cursor: not-allowed;
}

.emotion-34:disabled::-webkit-input-placeholder {
  color: #909296;
}

.emotion-34:disabled::-moz-placeholder {
  color: #909296;
}

.emotion-34:disabled:-ms-input-placeholder {
  color: #909296;
}

.emotion-34[data-disabled]::-webkit-input-placeholder {
  color: #909296;
}

.emotion-34[data-disabled]::-moz-placeholder {
  color: #909296;
}

.emotion-34[data-disabled]:-ms-input-placeholder {
  color: #909296;
}

.emotion-34:disabled::placeholder,
.emotion-34[data-disabled]::placeholder {
  color: #909296;
}

.emotion-34[data-invalid] {
  color: #fa5252;
  border-color: #fa5252;
}

.emotion-34[data-invalid]::-webkit-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-34[data-invalid]::-moz-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-34[data-invalid]:-ms-input-placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-34[data-invalid]::placeholder {
  opacity: 1;
  color: #fa5252;
}

.emotion-34[data-with-icon] {
  padding-left: 2.25rem;
}

.emotion-34::-webkit-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-34::-moz-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-34:-ms-input-placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-34::placeholder {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
  opacity: 1;
}

.emotion-34::-webkit-inner-spin-button,
.emotion-34::-webkit-outer-spin-button,
.emotion-34::-webkit-search-decoration,
.emotion-34::-webkit-search-cancel-button,
.emotion-34::-webkit-search-results-button,
.emotion-34::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.emotion-34[type=number] {
  -moz-appearance: textfield;
}

.emotion-34[data-read-only] {
  cursor: default;
}

.emotion-36 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #adb5bd;
}

.emotion-38 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 1rem;
  margin-top: 1.5rem;
}

.emotion-38>* {
  box-sizing: border-box;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.emotion-41 {
  -webkit-tap-highlight-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 1rem;
  background-color: transparent;
  text-align: left;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 2.25rem;
  padding-left: 1.125rem;
  padding-right: 1.125rem;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 0.25rem;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 0.875rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 0.0625rem solid #228be6;
  background-color: transparent;
  color: #228be6;
}

.emotion-41:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-41:focus:not(:focus-visible) {
  outline: none;
}

.emotion-41:focus {
  outline-offset: 0.125rem;
  outline: 0.125rem solid #339af0;
}

.emotion-41:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-41:hover {
    background-color: rgba(231, 245, 255, 0.35);
  }
}

@media (hover: none) {
  .emotion-41:active {
    background-color: rgba(231, 245, 255, 0.35);
  }
}

.emotion-41:active {
  -webkit-transform: translateY(0.0625rem);
  -moz-transform: translateY(0.0625rem);
  -ms-transform: translateY(0.0625rem);
  transform: translateY(0.0625rem);
}

.emotion-41:disabled,
.emotion-41[data-disabled] {
  border-color: transparent;
  background-color: #e9ecef;
  color: #adb5bd;
  cursor: not-allowed;
  background-image: none;
  pointer-events: none;
}

.emotion-41:disabled:active,
.emotion-41[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-41[data-loading] {
  pointer-events: none;
}

.emotion-41[data-loading]::before {
  content: "";
  position: absolute;
  top: -0.0625rem;
  right: -0.0625rem;
  left: -0.0625rem;
  bottom: -0.0625rem;
  background-color: rgba(255, 255, 255, .5);
  border-radius: 0.25rem;
  cursor: not-allowed;
}

.emotion-42 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-44 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
    className="emotion-0"
  >
    <div
      className="emotion-1 emotion-2 emotion-3"
    >
      <label
        className="emotion-4 emotion-5 emotion-6"
        htmlFor="mantine-vup1fqgb5"
        id="mantine-vup1fqgb5-label"
        onMouseDown={[Function]}
      >
        Flight Number
        <span
          aria-hidden={true}
          className="emotion-7 emotion-8 emotion-9"
        >
           *
        </span>
      </label>
      <div
        className="emotion-10 emotion-11 emotion-12"
      >
        <input
          aria-invalid={false}
          className="emotion-13 emotion-14 emotion-15"
          id="mantine-vup1fqgb5"
          onChange={[Function]}
          placeholder="Flight Number"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-1 emotion-17 emotion-18"
    >
      <label
        className="emotion-4 emotion-20 emotion-6"
        htmlFor="mantine-o2jnbrpfe"
        id="mantine-o2jnbrpfe-label"
        onMouseDown={[Function]}
      >
        Departure Date
        <span
          aria-hidden={true}
          className="emotion-7 emotion-8 emotion-24"
        >
           *
        </span>
      </label>
      <div
        className="emotion-10 emotion-26 emotion-12"
      >
        <div
          className="emotion-28 emotion-29 emotion-30"
        >
          <svg
            className="tabler-icon tabler-icon-calendar"
            fill="none"
            height="1.5rem"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            width="1.5rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 7a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v12a2 2 0 0 1 -2 2h-12a2 2 0 0 1 -2 -2v-12z"
            />
            <path
              d="M16 3v4"
            />
            <path
              d="M8 3v4"
            />
            <path
              d="M4 11h16"
            />
            <path
              d="M11 15h1"
            />
            <path
              d="M12 15v3"
            />
          </svg>
        </div>
        <button
          aria-controls="mantine-2yieoj20t-dropdown"
          aria-expanded={false}
          aria-haspopup="dialog"
          aria-invalid={false}
          className="emotion-31 emotion-13 emotion-31 emotion-34"
          data-dates-input={true}
          data-with-icon={true}
          id="mantine-2yieoj20t-target"
          onClick={[Function]}
          required={true}
          type="button"
        >
          <span
            className="emotion-35 emotion-36"
          >
            Departure Date
          </span>
        </button>
      </div>
    </div>
    <input
      type="hidden"
      value=""
    />
    <div
      className="emotion-37 emotion-38"
    >
      <button
        className="emotion-39 emotion-40 emotion-41"
        data-button={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="emotion-42 emotion-43"
        >
          <span
            className="emotion-44 emotion-45"
            style={
              {
                "textTransform": undefined,
              }
            }
          >
            Get Availability
          </span>
        </div>
      </button>
    </div>
  </div>,
]
`;
